File: python/consumer.py
import kafka
import logging

logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.INFO)
brokerPort = "9092"


consumer = kafka.KafkaConsumer(
    'Users',
    bootstrap_servers=["127.0.0.1:" + brokerPort],
    auto_offset_reset='earliest',
    enable_auto_commit=True,
    group_id='test',
    value_deserializer=lambda x: x.decode('utf-8')
)

for message in consumer:
    logging.info(f'offset={message.offset}, key={message.key}, value={message.value}')

File: js/consumer.js
// consume message

const Kafka = require("kafkajs").Kafka
const getIpAdress = require("./host_ip.js");
const ipAddress = getIpAdress();
const brokerPort = "9092"

run();

async function run(){
    try {
        const kafka = new Kafka({
            "clientId": "admin",
            "brokers": [ipAddress.concat(":").concat(brokerPort)]
        })
        const consumer = kafka.consumer({"groupId": "test"});
        console.log("Connecting...")
        await consumer.connect()
        console.log("Connected!")

        consumer.subscribe({
            "topic": "Users",
            "fromBeginning": true
        })

        await consumer.run({
            "eachMessage": async result => {
                console.log(`Received message ${result.message.value} on partition ${result.partition}`)
            }
        })
    }
    catch(ex) {
        console.log("Error occurred.")
    }
    finally{
        
    }
}